src/univariate/continuous/landau.jl

```diff
@@ -0,0 +1,191 @@
+using StaticArrays
+"""
+    Landau(μ,θ)
+
+The *Landau distribution* with location `μ` and scale `θ` has probability density function
+
+```math
+p(x) = \\frac{1}{2 \\pi i} \\int_{\\mu-i\\infty}^{\\mu+i\\infty} e^{s \\log(s) + x s}\\, ds
+```
+
+```julia
+Landau()       # Landau distribution with zero location and unit scale, i.e. Landau(0, 1)
+Landau(u)      # Landau distribution with location u and unit scale, i.e. Landau(u, 1)
+Landau(u, b)   # Landau distribution with location u ans scale b
+
+params(d)       # Get the parameters, i.e. (u, b)
+location(d)     # Get the location parameter, i.e. u
+scale(d)        # Get the scale parameter, i.e. b
+```
+
+External links
+
+* [Landau distribution on Wikipedia](http://en.wikipedia.org/wiki/Landau_distribution)
+
+"""
+struct Landau{T<:Real} <: ContinuousUnivariateDistribution
+    μ::T
+    θ::T
+    Landau{T}(µ::T, θ::T) where {T} = new{T}(µ, θ)
+end
+
+
+function Landau(μ::T, θ::T; check_args=true) where {T <: Real}
+    check_args && @check_args(Landau, θ > zero(θ))
+    return Landau{T}(μ, θ)
+end
+
+Landau(μ::Real, θ::Real) = Landau(promote(μ, θ)...)
+Landau(μ::Integer, θ::Integer) = Landau(float(μ), float(θ))
+Landau(μ::T) where {T <: Real} = Landau(μ, one(T))
+Landau() = Landau(0.0, 1.0, check_args=false)
+
+@distr_support Landau -Inf Inf
+
+#### Conversions
+function convert(::Type{Landau{T}}, μ::S, θ::S) where {T <: Real, S <: Real}
+    Landau(T(μ), T(θ))
+end
+function convert(::Type{Landau{T}}, d::Landau{S}) where {T <: Real, S <: Real}
+    Landau(T(d.μ), T(d.θ), check_args=false)
+end
+
+#### Parameters
+
+location(d::Landau) = d.μ
+scale(d::Landau) = d.θ
+
+params(d::Landau) = (d.μ, d.θ)
+@inline partype(d::Landau{T}) where {T<:Real} = T
+
+
+#### Evaluation
+
+pdf(d::Landau, x::Real) = _landau_pdf((x-d.μ)/d.θ)
+
+cdf(d::Landau, x::Real) = _landau_cdf((x-d.μ)/d.θ)
+
+function cf(d::Landau, t::Real)
+    exp(im*t*d.μ - 2im*d.θ*t/π*log(abs(t)) - d.θ*abs(t))
+end
```

Comment:
The following definition uses `cis(x) = exp(im*x)` and avoids NaN values for `t = 0`:
```suggestion
function cf(d::Landau, t::Real)
    z = t * (d.μ + d.θ * (sign(t) * im - log(abs(t)) * 2 / π))
    return iszero(t) ? cis(zero(z)) : cis(z) 
end
```